src/number_theory/von_mangoldt.lean

```diff
@@ -0,0 +1,121 @@
+/-
+Copyright (c) 2022 Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bhavik Mehta
+-/
+
+import algebra.is_prime_pow
+import number_theory.arithmetic_function
+import analysis.special_functions.log
+
+/-!
+# The von Mangoldt Function
+
+In this file we define the von Mangoldt function: the function on natural numbers that returns
+`log p` if the input can be expressed as `p^k` for a prime `p`.
+
+## Main Results
+
+The main definition for this file is
+
+- `nat.arithmetic_function.von_mangoldt`: The von Mangoldt function `Λ`.
+
+We then prove the classical summation property of the von Mangoldt function in
+`nat.arithmetic_function.von_mangoldt_sum`, that `∑ i in n.divisors, Λ i = real.log n`, and use this
+to deduce alternative expressions for the von Mangoldt function via Möbius inversion, see
+`nat.arithmetic_function.sum_moebius_mul_log_eq`.
+
+## Notation
+
+We use the standard notation `Λ` to represent the von Mangoldt function.
+
+-/
+
+namespace nat
+namespace arithmetic_function
+
+open finset
+
+open_locale arithmetic_function
+
+/-- `log` as an arithmetic function `ℕ → ℝ`. Note this is in the `nat.arithmetic_function`
+namespace to indicate that it is bundled as an `arithmetic_function` rather than being the usual
+real logarithm. -/
+noncomputable def log : arithmetic_function ℝ :=
+⟨λ n, real.log n, by simp⟩
+
+@[simp] lemma log_apply {n : ℕ} : log n = real.log n := rfl
+
+/-- In the case when `n` is a prime power, `min_fac` will give the appropriate prime. -/
```

Comment:
Please write a more helpful docstring, like you did in the module doc above.

Reply:
Yikes I have no idea how this docstring got here! Thanks for the catch!